rest-service/manager_rest/resource_manager.py

```diff
@@ -191,6 +191,30 @@ def restore_snapshot(self,
         )
         return execution
 
+    def _is_valid_yaml_file(self, plugin):
+        """Is the plugin YAML file valid?"""
+
+        with open(plugin.yaml_file_path()) as f:
+            plugin_yaml = yaml.safe_load(f)
+
+        plugins = plugin_yaml.get(constants.PLUGINS, {})
+        if not plugin_yaml.get(constants.PLUGINS, {}):
+            raise manager_exceptions.InvalidPluginError(
+                'YAML file must be in correct plugin format.'
+            )
+        for plugin_spec in plugins.values():
+            if plugin_spec.get(constants.PLUGIN_PACKAGE_NAME) == \
```

Comment:
You can make this shorter by switching the `if` to `if  not` and move the `return true` to the end of the method